### DISCUSSION TITLE
Different number of input images in the output montage?

### DISCUSSION THREAD

**Discussion by eejap** [ImageMagick/ImageMagick#5253] [Help] [answered]
Hi there. I am trying to create a montage of images of equal size but with different numbers of these images on different lines, say 3 on the first line then 4 on the next, then 2 on the final line.

I would also preferably like to call these images from a separate text file. So far this text file (here images.txt) looks something like this:

image1.png image2.png image3.png \
image4.png image5.png image6.png image7.png \
image8.png image9.png

With my montage command looking something like:
montage @image.txt -tile 4x3 -geometry +0+0 -title 'My Plot' ordered.png

images.txt is in the same directory as the image pngs

Thanks!

**Comment by fmw42** [comment, marked as answer]
Montage will not allow different numbers per row. But you can add null: placeholders to fill out -tile 4x3.  

```
image1.png image2.png image3.png null:
image4.png image5.png image6.png image7.png 
image8.png image9.png null: null:

```
So in Unix syntax:

```
montage image1.png image2.png image3.png null: \
image4.png image5.png image6.png image7.png \
image8.png image9.png null: null: \
 -tile 4x3 -geometry +0+0 -title 'My Plot' ordered.png

```

or you can do 3 appends for the rows and then append all the rows

```
convert \
\( image1.png image2.png image3.png +append \) \
\( image4.png image5.png image6.png image7.png +append \) \
\( image8.png image9.png +append \) \
-background white -append \
ordered.png

```

**Reply by eejap** [reply]
That's great, thank you so much!
